### DISCUSSION TITLE
Memory limits on the ABit AB-AV4 (aka VL-BUS 486)

### DISCUSSION THREAD

**Discussion by ppgrainbow** [86Box/86Box#1346] [Q&A]
I have found information regarding the ABit AB-AV4 and I wondering if it would be okay to provide a fix to correct the memory limits on the AB-AV4 (also known as VL-BUS 486).

Checking line 213 of machine_table.c:

 { "[SiS 460] ABIT AV4",			"av4",			MACHINE_TYPE_486,		CPU_PKG_SOCKET3, 0, 0, 0, 0, 0, 0, 0,										MACHINE_VLB | MACHINE_IDE,							 1024, 65536, 1024, 127,		  machine_at_av4_init, NULL			},

A real ABit AB-AV4 can only support up to 32 MB of RAM, not 64 MB as noted in this link: http://www.win3x.org/uh19/motherboard/show/79

32 MB of RAM can be achieved with (4) 4M x 9 on Bank 0 and Bank 1.

I have provided a quick correction in line 213 of machine_table.c for this that I can forward to OBattler:

 { "[SiS 460] ABit AB-AV4 (VL-BUS 486)",	"av4",			MACHINE_TYPE_486,		CPU_PKG_SOCKET3, 0, 0, 0, 0, 0, 0, 0,										MACHINE_VLB | MACHINE_IDE,							 1024, 32768, 1024, 127,		  machine_at_av4_init, NULL			},

Thank you for your time and have a good day. :)

**Comment by dgurney** [comment]
Are you sure 32MB is the real hard limit? It's not unheard of for the spec sheet and reality to disagree with each other.

**Reply by ppgrainbow** [reply]
I am truely sure that 32 MB is the hard limit on the Abit AB-AV4 VL-Bus motherboard. The DRAM configuration is used to determine how much memory that the motherboard will handle on Banks 0 and Banks 1. Go through this page and see for yourself: https://stason.org/TULARC/pc/motherboards/A/ABIT-COMPUTER-CORPORATION-486-AV4-VL-BUS-MAIN-BOAR.html

The BIOS string on the bottom of the POST screen, "40-00AG-001247-00101111-060692-SIS3486" matches to to the one that is emulated machine:

![20210331-222108-480](https://user-images.githubusercontent.com/32658796/113218957-4a282900-9235-11eb-86de-f5e59f11b492.png)

Despite the SiS 461 chipset itself being limited to 64 MB, some of the motherboards in question only have two memory banks, Banks 0 and Banks 1, limiting the amount of memory on the motherboard to 32 MB.

For a motherboard with a SiS 461 chipset to achieve 64 MB, four memory banks, Banks 0 through Bank 3 have to be present and the AB-AV4 only has Banks 0 and 1. I truly don't think that you can put a 16M x 9 memory module in Bank 0 and/or Bank 1 as the motherboard can't support it.

So, with (4) 4M x 9 DRAM modules on Banks 0 and 1, the memory limit on the AB-AV4 is 32 MB.

**Reply by Deksor** [reply]
Hello ! I was involved in the creation of Ultimate Hardware 2019 and it's based on TH99/Stason.org, so quoting stason.org is like quoting an old version of UH19 :)

However you're correct, this board uses 30 pin SIMMS which were mostly available in 256K/1M/4M styles. This board has 8 slots so 8*4 = 32MB (bigger capacities existed, but boards rarely supported them). The chipset may support more, but you can't physically install more than 32MB on that board.

**Reply by ppgrainbow** [reply]
I'm with you on this one, Dekor. :)

Let's hope that the memory limits on the ABit AB-AV4 gets corrected. ^_^
